orbit/tools/build/build-windows.go

```diff
@@ -0,0 +1,265 @@
+	"github.com/fleetdm/fleet/v4/orbit/pkg/constant"
+	"github.com/josephspurrier/goversioninfo"
+	zlog "github.com/rs/zerolog/log"
+)
+
+func main() {
+	// Input flags
+	flagVersion := flag.String("version", "0.0.1", "Version string")
+	flagCreateResource := flag.Bool("resource", false, "This is a bool flag to just create the resource.syso file and not build the binary")
+	flagIcon := flag.String("icon", "windows_app.ico", "Path to the icon file to embedded on the binary")
+	flagOutputBinary := flag.String("output", "output.exe", "Path to the output binary")
+	flagCmdDir := flag.String("input", "", "Path to the directory containing the utility to build")
+
+	flag.Usage = func() {
+		zlog.Fatal().Msgf("Usage: %s -version <version> -input <dir_path> -output <output_binary>\n", os.Args[0])
+	}
+	flag.Parse()
+
+	// checking input flags are not empty
+	if *flagCmdDir == "" {
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	// check if flagCmdDir is a valid directory
+	_, err := os.Stat(*flagCmdDir)
+	if err != nil {
+		zlog.Fatal().Err(err).Msg("invalid cmd-dir")
+		flag.Usage()
+		os.Exit(1)
+	}
+
+	// now we need to create the 'resource.syso' metadata file which contains versioninfo data
+
+	// lets start with sanitizing the version data
+	vParts, err := sanitizeVersion(*flagVersion)
+	if err != nil {
+		zlog.Fatal().Err(err).Msgf("invalid version: %s", *flagVersion)
+		os.Exit(1)
+	}
+
+	// then we need to create the manifest.xml file
+	manifestPath, err := writeManifestXML(vParts, *flagCmdDir)
+	if err != nil {
+		zlog.Fatal().Err(err).Msg("creating manifest.xml")
+		os.Exit(1)
+	}
+	if !*flagCreateResource {
+		defer os.Remove(manifestPath)
+	}
+
+	// now we can create the VersionInfo struct
+	targetIconPath := *flagCmdDir + "/" + *flagIcon
+	vi, err := createVersionInfo(vParts, targetIconPath, manifestPath)
+	if err != nil {
+		zlog.Fatal().Err(err).Msg("parsing versioninfo")
+		os.Exit(1)
+	}
+
+	// and finally we can write the 'resource.syso' file
+	vi.Build()
+	vi.Walk()
+
+	// resource.syso is the resource file that is going to be picked up by golang compiler
+	outPath := filepath.Join(*flagCmdDir, "resource.syso")
+	if err := vi.WriteSyso(outPath, "amd64"); err != nil {
+		zlog.Fatal().Err(err).Msg("creating syso file")
+		os.Exit(1)
+	}
+
+	// checking if build should be performed or not
+	if *flagCreateResource {
+		zlog.Info().Msgf("Successfully created resource file %s", outPath)
+	} else {
+
+		defer os.Remove(outPath)
+		// now we can build the binary
+		if err := buildTargetBinary(*flagCmdDir, *flagVersion, *flagOutputBinary); err != nil {
+			zlog.Fatal().Err(err).Msg("error building binary")
+			os.Exit(1)
+		}
+
+		zlog.Info().Msgf("Successfully built %s", *flagOutputBinary)
+	}
+}
+
+// createVersionInfo returns a VersionInfo struct pointer to be used to generate the 'resource.syso'
+// metadata file (see writeResourceSyso).
+func createVersionInfo(vParts []string, iconPath string, manifestPath string) (*goversioninfo.VersionInfo, error) {
```

Comment:
This seems to be a duplicate of https://github.com/fleetdm/fleet/pull/10783/files#diff-95043ac955352f5e6fe86545b3a5cebae5caa643a9e91da888d302e6d1c7fb98R263 ?

Reply:
although they have the same name, there are some minor differences between the functions, such as the file description information and the ability to embed an icon as part of the versioninfo resource (new parameter on the function signature). We might want to keep these two separate for the time being as there could be different customizations for the orbit files (such as locale support) and the installer file